tests/test_nightReporter.py

```diff
@@ -0,0 +1,137 @@
+# This file is part of rapid_analysis.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Test cases for the NightReporter."""
+
+import unittest
+import tempfile
+from unittest import mock
+
+import lsst.utils.tests
+
+import matplotlib as mpl
+mpl.use('Agg')
+
+from lsst.rapid.analysis.nightReport import NightReporter, loadReport, saveReport  # noqa: E402
+import lsst.rapid.analysis.butlerUtils as bu  # noqa: E402, N813
+
+import lsst.daf.butler as dafButler  # noqa: E402
+NOBUTLER = True  # noqa: E402
+try:  # noqa: E402
+    butler = dafButler.Butler('LATISS')  # noqa: E402
+    assert isinstance(butler, dafButler.Butler)  # noqa: E402
+    # don't leak this out! self.butler is used elsewhere, and if butler
+    # is defined in scope this can cause bugs in the tests themselves
+    del butler  # noqa: E402
+    NOBUTLER = False  # noqa: E402
+except (FileNotFoundError):  # noqa: E402
+    print("No LATISS butler found, skipping butler-driven tests.")  # noqa: E402
```

Comment:
What is the purpose of all these module-level globals? This could all live in setupClass, and then call `self.skip` there.

Reply:
Yup, good shout, thanks. All moved now 👍